### PR TITLE
feat: migrate Tile component to Fiori3

### PR DESCRIPTION
**Pull request by InnaAtanasova**
## Related Issue
Closes SAP/fundamental-styles#897
Closes SAP/fundamental-styles#898
Closes SAP/fundamental-styles#899
Closes SAP/fundamental-styles#993

## Description
Migrate Tile component to Fiori3
Introduce Numeric Content, KPI Tiles, Launch Tiles, Feed Tiles, Slide Tile
Add Action mode to the tiles

**BREAKING CHANGES**
Removed classes:
- `fd-tile__media`
- `fd-tile__text`
- `fd-tile__actions`
- `fd-product-tile` - `fd-product-tile__media`, `fd-product-tile__content`, `fd-product-tile__title` and `fd-product-tile__text`
- `fd-tile__title` is not  a child of `fd-tile__content`

### CODE REVIEW

File: docs/pages/components/tile.md [hunk 1]
```diff
@@ -10,186 +10,637 @@ summary:
...
+<br>
+Numeric Content Sizes:
+<ul class="docs-ul">
+    <li>L-Numeric Content size. Longest combination for 4-character languages</li>
+    <li>M-Numeric Content size. Longest combination for 6-character languages</li>
+    <li>S-Numeric Content size. Longest combination for 8-character languages</li>
+</ul>
 
 <br>
+The header contains a maximum of 3 lines title or a maximum of 2 lines title with 1 line subtitle.
```
Comment (Vanessa-Cusmich) [inline docs/pages/components/tile.md:236]: The header can contain a maximum of 3 lines; all 3 lines can be used for the title or one of the lines can be used for a subtitle. If a 4 line title or an optional subtitle line is required, the `fd-tile_header-content` can be used to represent a header and will merge the content containers into one tile.

File: docs/pages/components/tile.md [hunk 2]
```diff
@@ -10,186 +10,637 @@ summary:
...
 {% endcapture %}
-{% include display-component.html component=tile %}
+{% include display-component.html component=tile-launch %}
 
 <br>
 
-## Disabled state
-Add class `is-disabled`
+## KPI Tile (Numeric Content)
+Key Performance Indicators (KPIs) are used to measure and monitor a company’s performance at a strategic and operational level.The number of digits is limited by the size of the tiles.
```
Comment (Vanessa-Cusmich) [inline docs/pages/components/tile.md:390]: The KPI Tile control supports a limited number of digits which varies depending on the size of the tile.

**Comment by JKMarkowski** [comment]
image

Text overflow on tiles doesn't work properly. Let's have a talk about it, because there is no way to handle it on pure css, without hardcoding fixed height for every single element.

File: src/tile.scss [hunk 3]
```diff
@@ -2,96 +2,278 @@
...
+
+    &.#{$block}--s {
+      .#{$block}__content {
+        .#{$block}__section {
+          &:nth-child(2) {
+            max-width: 8.25rem;
+          }
+        }
+      }
+    }
```
Comment (JKMarkowski) [inline src/tile.scss:277]: Is there any certain reason why we have it ?

File: src/numeric-content.scss [hunk 4]
```diff
@@ -0,0 +1,247 @@
+@import "./settings";
+@import "./mixins";
+@import "./functions";
```
Comment (JKMarkowski) [inline src/numeric-content.scss:3]: Could you check and remove unused imports ?

File: src/tile.scss [hunk 5]
```diff
@@ -2,96 +2,278 @@
...
+    // box-shadow: 0 0 0 1px fade(var(--sapContent_ShadowColor), 30);;
+  }
+
+  @include fd-focus() {
+    box-shadow: none;
+    outline-style: dotted;
+    outline-width: var(--sapContent_FocusWidth);
+    outline-color: var(--sapContent_FocusColor);
+    outline-offset: -0.1875rem;
   }
```
[the remark this answers is not in the document]
Comment (InnaAtanasova) [reply src/tile.scss:62]: oops :) I actually didn't use this. I forgot to delete it. Thanks for noticing!

File: src/numeric-content.scss (hunk 4 above)
Comment (InnaAtanasova) [reply src/numeric-content.scss:3]: I will! Thanks for pointing out!

**Comment by prsdthkr** [comment]
> Text overflow on tiles doesn't work properly. Let's have a talk about it, because there is no way to handle it on pure css, without hardcoding fixed height for every single element.

Should tiles be allowed to have flexible height based on inner content? This would be better for responsiveness too.

**Comment by InnaAtanasova** [comment]
> > Text overflow on tiles doesn't work properly. Let's have a talk about it, because there is no way to handle it on pure css, without hardcoding fixed height for every single element.
> 
> Should tiles be allowed to have flexible height based on inner content? This would be better for responsiveness too.

Hi @prsdthkr , no the height and the width of the tiles are fixed - 11rem x 11rem for Large and 9.25rem x 9.25rem for small tile. The same is valid for the height of the sections of the Tile. Header - 3.625rem (Small Tile - 3.25rem), Content - 3.875rem (Small Tile - 3.438 rem) and Footer - 1rem (Small Tile - 1.063 rem)

File: docs/pages/components/tile.md [hunk 6]
```diff
@@ -10,186 +10,727 @@ summary:
...
+{% include display-component.html component=tile-feed %}
 
 <br>
 
-<div class="fd-tile is-disabled" aria-disabled="true">
-    <div class="fd-tile__media">
-        <span class="fd-avatar fd-avatar--m fd-avatar--transparent sap-icon--home"></span>
+## Badge Tile
+Badges are used to display very short and important information.
+Use a badge on a Tile if the Tile new or it has been newly added.
```
Comment (Vanessa-Cusmich) [inline docs/pages/components/tile.md:651]: Use a badge on a Tile if the Tile is new or has been newly added to a page.

**Comment by JKMarkowski** [comment]
Let's increase badge's z-index
image

File: src/tile.scss [hunk 7]
```diff
@@ -1,97 +1,657 @@
...
+      @include fd-flex($direction: column) {
+        justify-content: space-between;
+      }
+
+      z-index: 2;
+      width: 100%;
+      max-width: 100%;
+      background-color: var(--sapBlockLayer_Background);
+      opacity: 0.8;
+      padding-top: 0.375rem;
```
Comment (JKMarkowski) [inline src/tile.scss:460]: Shouldn't it be `0.25rem`?
image
image

File: src/tile.scss [hunk 8]
```diff
@@ -1,97 +1,657 @@
...
+      .#{$block}__container {
+        @include set-height(5.25rem);
+
+        padding-top: 0.25rem;
+        padding-right: $fd-tile-small-padding;
+        padding-left: $fd-tile-small-padding;
+      }
+
+      .#{$block}__header {
+        @include set-height(3.188rem);
```
Comment (JKMarkowski) [inline src/tile.scss:522]: It shouldn't be 3.25rem ?
image
image

File: src/tile.scss [hunk 9]
```diff
@@ -1,97 +1,657 @@
...
+      opacity: 0.8;
+      padding-top: 0.375rem;
+      padding-left: $fd-tile-padding;
+      padding-right: $fd-tile-padding;
+      border-bottom-left-radius: $fd-tile-border-radius;
+      border-bottom-right-radius: $fd-tile-border-radius;
+    }
+
+    .#{$block}__header {
+      @include set-height(3.5rem);
```
Comment (JKMarkowski) [inline src/tile.scss:468]: Also this (3.625rem)
image
image

**Comment by InnaAtanasova** [comment]
> Let's increase badge's z-index
> image

Nice catch. Thank you for noticing it

**Comment by InnaAtanasova** [comment]
> * [ ]  Firefox: Strange focus on x button
>   image
> * [ ]  Firefox: hover behaviour
>   image

The first one is coming from the fact that the button has a bigger touchable area and the visible part of the button is positioned in the top right corner. I used the fake-focus approach to fix it. 

Specs:
Screen Shot 2020-05-20 at 12 57 50 PM

Now:
Screen Shot 2020-05-20 at 12 59 29 PM


Second comment:
It's per specs
Screen Shot 2020-05-20 at 1 00 06 PM

**Comment by stefanoScalzo** [comment]
Screen Shot 2020-05-20 at 3 36 32 PM
Screen Shot 2020-05-20 at 3 36 41 PM
Screen Shot 2020-05-20 at 3 36 47 PM
 Unnormalized tests show that we need to reset

**Comment by stefanoScalzo** [comment]
Screen Shot 2020-05-20 at 3 48 03 PM
Screen Shot 2020-05-20 at 3 48 07 PM

The black background is too light compared to specs

**Comment by stefanoScalzo** [comment]
The size of the x should be Size: 1.375 rem*1.375 rem;
but you have it as 2rem-0.375 with  which give 1.625 and not 1.375

**Comment by InnaAtanasova** [comment]
> Screen Shot 2020-05-20 at 3 48 03 PM
> Screen Shot 2020-05-20 at 3 48 07 PM
> 
> The black background is too light compared to specs

@stefanoScalzo 
The specs:
Screen Shot 2020-05-20 at 4 48 56 PM

The code:
Screen Shot 2020-05-20 at 4 50 01 PM

**Comment by InnaAtanasova** [comment]
> Screen Shot 2020-05-20 at 3 36 32 PM
> Screen Shot 2020-05-20 at 3 36 41 PM
> Screen Shot 2020-05-20 at 3 36 47 PM
> Unnormalized tests show that we need to reset

Can't reproduce it locally

File: docs/pages/components/tile.md [hunk 10]
```diff
@@ -10,186 +10,911 @@ summary:
...
+                <span class="fd-tile__dot"></span>
+                <span class="fd-tile__dot"></span>
+            </div>
+        </div>
+    </div>
+</div>
+<div class="docs-section-container">
+    <div tabindex="0" class="fd-tile fd-tile--double fd-tile--slide">
+        <div class="fd-tile__background-img" style="background-image: url('https://placeimg.com/400/200/nature')"></div>
+        <<button aria-label="toggle pause button" class="fd-tile__toggle fd-tile__toggle--pause"></button>
```
Comment (prsdthkr) [inline docs/pages/components/tile.md:782]: Extra `<` causing:
image

File: docs/pages/components/tile.md [hunk 11]
```diff
@@ -10,186 +10,911 @@ summary:
...
+                <span class="fd-tile__dot"></span>
+                <span class="fd-tile__dot"></span>
+                <span class="fd-tile__dot"></span>
+                <span class="fd-tile__dot"></span>
+            </div>
+        </div>
+    </div>
+    <div tabindex="0" class="fd-tile fd-tile--double fd-tile--s fd-tile--slide">
+        <div class="fd-tile__background-img" style="background-image: url('https://placeimg.com/400/200/nature')"></div>
+        <<button aria-label="toggle pause button" class="fd-tile__toggle fd-tile__toggle--pause"></button>
```
Comment (prsdthkr) [inline docs/pages/components/tile.md:804]: Extra `<`.

File: docs/pages/components/tile.md [hunk 12]
```diff
@@ -10,186 +10,911 @@ summary:
...
+        <div class="fd-tile__container">
+            <div class="fd-tile__header">
+                <div class="fd-tile__title">Wind Map: Monitoring Real-Time and Forecasted Wind Conditions across the Globe</div>
+                <div class="fd-tile__subtitle">SAP News</div>
+            </div>
+            <div class="fd-tile__footer">
+                <span class="fd-tile__footer-text">May 14, 2020</span>
+            </div>
```
Comment (prsdthkr) [inline docs/pages/components/tile.md:812]: All this text should have `text-shadow` for accessibility (color contrast) reasons. This is also according to spec:
image

Comment (InnaAtanasova) [reply docs/pages/components/tile.md:812]: Hi @prsdthkr the font color and the text-shadow use calculations which are not themable. We've contacted the team working on the sap-theming and we also logged an issue https://github.com/SAP/fundamental-styles/issues/1048

File: docs/pages/components/tile.md (hunk 10 above)
Comment (InnaAtanasova) [reply docs/pages/components/tile.md:782]: Thanks for catching this. I will fix it

File: docs/pages/components/tile.md (hunk 11 above)
Comment (InnaAtanasova) [reply docs/pages/components/tile.md:804]: https://github.com/SAP/fundamental-styles/pull/1054

File: docs/pages/components/tile.md (hunk 10 above)
Comment (InnaAtanasova) [reply docs/pages/components/tile.md:782]: https://github.com/SAP/fundamental-styles/pull/1054

**Pull request merged**